kernel/unwind.c

```diff
@@ -505,7 +522,11 @@ static int find_fde(uint64_t rip, eh_cie* cie_out, eh_fde* fde_out) {
     memset(&fde, 0, sizeof(eh_fde));
     fde.cie = &current_cie;
     if (read_fde(&subhead, &fde)) { return -1; }
-    if (rip >= fde.pc_begin && rip - fde.pc_begin < fde.pc_len) {
+    /* Offset our check by one instruction: it's unlikely that we're returning
```

Comment:
As we discussed, this logic is still slightly wrong for interrupt handlers. We can fault on the first instruction in a function.
Luckily for us, interrupt handlers are never noreturn, in a sense - with the exception of double-fault and exit(), they are /always/ called with the intention of actually returning to the instruction on the stack. So if we can figure out how to communicate that info into the eh_frame section, the interrupt handlers can use the more sane looking FDE check and not worry about falling off the end of any functions.